.github/workflows/ci.yml

```diff
@@ -58,12 +58,28 @@ jobs:
   build:
     strategy:
       matrix:
-        os: [macos-latest, ubuntu-latest, windows-latest]
+        target: [x86_64-unknown-linux-gnu, armv7-unknown-linux-gnueabihf, x86_64-apple-darwin, x86_64-pc-windows-gnu]
         include:
-          - os: ubuntu-latest
-            e2e: true
-          - os: macos-latest
-            e2e: true
+
+          - target: x86_64-unknown-linux-gnu
+            os: ubuntu-latest
+            test_features: testcontainers
+
+          - target: armv7-unknown-linux-gnueabihf
+            os: ubuntu-latest
+            # This target is only for cross-compiling, skip all tests.
+            skip_unit_tests: true
+            skip_e2e_test: true
+
+          - target: x86_64-apple-darwin
+            os: macos-latest
+
+          - target: x86_64-pc-windows-gnu
+            os: windows-latest
+            skip_e2e_test: true # Our e2e test framework doesn't work on windows.
+            skip_unit_tests: true # Can't run linux docker images on windows.
```

Comment:
We run e2e tests without docker on windows now: https://github.com/comit-network/comit-rs/pull/3322